### ISSUE TITLE
Declare/formalize a model file format

### ISSUE THREAD

**Issue by NickleDave** [vocalpy/vak#673]
#672 describes replacing several options in the config file with a single option `model_path`.
The semantics of the `model_path` option are that it points to a directory generated by `vak train`. We should formalize this as a file format, ideally baked into the code, e.g. as an attrs class.

Since we don't have existing constraints we can put this in place now to make life easier later--I would rather capture as much info as possible, in a directory, about model provenance, training procedure, etc, instead of needing to put a lot of that into code (like the torchvision `weights` abstraction).

Currently the "file format" is:
- directory
- with log file
- config file
- checkpoints folder
  - that has "intermittent save" checkpoint as well as "best performance on validation set" checkpoint
- parts of transform pipeline, e.g. SpectScaler -- could we save the whole transform pipeline?
- for frame classification: labelmap.json
